File: JavaLox/src/com/craftinginterpreters/lox/Array.java
package com.craftinginterpreters.lox;

import java.util.List;

/**
 *
 * @author brusc
 */
class Array implements LoxCallable {
   @Override
   public int arity() {
      return 1;
   }

   @Override
   public Object call(Interpreter interpreter, List<Object> arguments) {
      int size = (int) (double) arguments.get(0);
      return new LoxArray(size);
   }
}

File: CsLox/com/craftinginterpreters/lox/Array.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    internal class Array : LoxCallable
    {
        public int arity()
        {
            return 1;
        }

        public Object call(Interpreter interpreter, List<Object> arguments)
        {
            int size = (int)(double)arguments[0];
            return new LoxArray(size);
        }
    }
}
